components/health_controller/src/reporters.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright 2024 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::{
+    sync::Arc,
+    time::{Duration, Instant},
+};
+
+use kvproto::pdpb;
+use pdpb::SlowTrend as SlowTrendPb;
+use prometheus::IntGauge;
+
+use crate::{
+    slow_score::{SlowScore, SlowScoreTickResult},
+    trend::{RequestPerSecRecorder, Trend},
+    HealthController, HealthControllerInner, RaftstoreDuration,
+};
+
+pub struct RaftstoreReporterConfig {
+    pub inspect_interval: Duration,
+
+    pub unsensitive_cause: f64,
+    pub unsensitive_result: f64,
+    pub net_io_factor: f64,
+
+    pub cause_spike_filter_value_gauge: IntGauge,
+    pub cause_spike_filter_count_gauge: IntGauge,
+    pub cause_l1_gap_gauges: IntGauge,
+    pub cause_l2_gap_gauges: IntGauge,
+
+    pub result_spike_filter_value_gauge: IntGauge,
+    pub result_spike_filter_count_gauge: IntGauge,
+    pub result_l1_gap_gauges: IntGauge,
+    pub result_l2_gap_gauges: IntGauge,
+}
```

Comment:
Better to have comments to desribe these fieds or related links about the design.

Reply:
IMO, the params related with SlowTrend can directly refer the definition in `components/tikv_utils/src/trend.rs`:

https://github.com/tikv/tikv/blob/5d190fa5c2af88e0cd643aa6368383bb22a83241/components/tikv_util/src/trend.rs#L412-L431

Reply:
IMO, these params to SlowTrend will be simplified in the later work. Now, just add the reference to the definition in `trend.rs` is good to me.